### PR TITLE
Ap/aliases completer

### PR DESCRIPTION
**Pull request by jnoortheen**
This adds generic attribute check. So no need for instance checks https://github.com/xonsh/xonsh/pull/4267#discussion_r676066853

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: xonsh/cli_utils.py [hunk 1]
```diff
@@ -371,6 +373,149 @@ def dispatch(parser: ap.ArgumentParser, args=None, lenient=False, **ns):
...
+
+    def __init__(self, parser: ap.ArgumentParser, command, **kwargs):
+        args = tuple(c.value for c in command.args[: command.arg_index])
+
+        self.parser, self.remaining_args = self.get_parser(parser, args[1:])
+
+        self.command = command
+        kwargs["command"] = command
+        self.kwargs = kwargs
+        """will be sent to completer function"""
```
Comment (gforsyth) [inline xonsh/cli_utils.py:387]: should this line be here or did it get cut from somewhere else?

**Review by gforsyth — COMMENTED** [review]
Looks good @jnoortheen -- just one comment on a line that looks like it belongs elsewhere

File: xonsh/cli_utils.py (hunk 1 above)
Comment (jnoortheen) [reply xonsh/cli_utils.py:387]: the comment is correct. The kwargs will be used to call the completer functions

Comment (gforsyth) [reply xonsh/cli_utils.py:387]: Ok, can you make it a comment and not a string then, please?

Comment (jnoortheen) [reply xonsh/cli_utils.py]: done.

**Review by gforsyth — APPROVED** [review]

**Comment by gforsyth** [comment]
Thanks for this, @jnoortheen and sorry for the delay in getting it in!

**Pull request merged**